pkg/loki/modules.go

```diff
@@ -404,7 +404,7 @@ func (t *Loki) initQuerier() (services.Service, error) {
 	toMerge := []middleware.Interface{
 		httpreq.ExtractQueryMetricsMiddleware(),
 		httpreq.ExtractQueryTagsMiddleware(),
-		httpreq.PropagateHeadersMiddleware(httpreq.LokiEncodingFlagsHeader),
+		httpreq.PropagateHeadersMiddleware(httpreq.LokiEncodingFlagsHeader, httpreq.LokiOriginalQueryResultsHeader),
```

Comment:
i think the querier change here only handles requests directly made to queriers http endpoint.

for requests that pass through schedulers, i believe we are encoding these headers within the request itself. 
for [protobuf encoding](https://github.com/grafana/loki/blob/79fd9203ebe434882d308a58a7b0b866f0509843/pkg/querier/queryrange/marshal.go#L364) and [json encoding](https://github.com/grafana/loki/blob/79fd9203ebe434882d308a58a7b0b866f0509843/pkg/querier/queryrange/codec.go#L612)

Reply:
ty, good catch